pkg/cloud/ovirt/providerIDcontroller/providerIDController.go

```diff
@@ -0,0 +1,150 @@
+package providerIDcontroller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	ovirtsdk "github.com/ovirt/go-ovirt"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog"
+	"k8s.io/klog/klogr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	"github.com/openshift/cluster-api-provider-ovirt/pkg/cloud/ovirt/clients"
+)
+
+var _ reconcile.Reconciler = &providerIDReconciler{}
+
+type providerIDReconciler struct {
+	log                  logr.Logger
+	client               client.Client
+	listNodesByFieldFunc func(key, value string) ([]corev1.Node, error)
+	fetchProviderIDFunc  func(string) (string, error)
+	ovirtApi             *ovirtsdk.Connection
+}
+
+func (r *providerIDReconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.log.WithValues("node", request.NamespacedName)
+	r.log.V(3).Info("Reconciling", request.NamespacedName)
+
+	// Fetch the Node instance
+	node := corev1.Node{}
+	err := r.client.Get(context.Background(), request.NamespacedName, &node)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		r.log.Error(err, "Failed getting node")
+		return reconcile.Result{}, fmt.Errorf("error getting node: %v", err)
+	}
+
+	if node.Spec.ProviderID != "" {
+		return reconcile.Result{}, nil
+	}
+
+	r.log.Info("spec.ProviderID is empty, fetching from ovirt")
+	id, err := r.fetchProviderIDFunc(node.Name)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed getting VM from oVirt: %v", err)
+	}
+
+	node.Spec.ProviderID = fmt.Sprintf("ovirt://%s", id)
+	err = r.client.Update(context.Background(), &node)
+	if err != nil {
+		klog.Errorf("Updating node %s failed %v", node.Name, err)
```

Comment:
You're mixing your logging libraries here, might be better to use `r.log` consistently

Reply:
thanks